File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/CategoryModule.java
package br.org.restaurantepopular.delivery.config;

import br.org.restaurantepopular.data_db.repository.category.CategoryDAO;
import br.org.restaurantepopular.data_db.repository.category.CategoryRepository;
import br.org.restaurantepopular.usecase.category.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CategoryModule {

    @Bean
    public GetCategoryByIdUseCase getCategoryByIdUseCase(
            ICategoryRepository categoryRepository) {
        return new GetCategoryByIdUseCase(categoryRepository);
    }

    @Bean
    public CreateCategoryUseCase createCategoryUseCase(
            ICategoryRepository categoryRepository) {
        return new CreateCategoryUseCase(categoryRepository);
    }

    @Bean
    public UpdateCategoryUseCase updateCategoryUseCase(
            ICategoryRepository categoryRepository) {
        return new UpdateCategoryUseCase(categoryRepository);
    }

    @Bean
    public ListCategoriesUseCase listCategoriesUseCase(
            ICategoryRepository categoryRepository) {
        return new ListCategoriesUseCase(categoryRepository);
    }

    @Bean
    public DeleteCategoryUseCase deleteCategoryUseCase(
            ICategoryRepository categoryRepository) {
        return new DeleteCategoryUseCase(categoryRepository);
    }

    @Bean
    public ICategoryRepository categoryRepository(CategoryDAO categoryDAO) {
        return new CategoryRepository(categoryDAO);
    }
}

File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/CategoryModule.kt
package br.org.restaurantepopular.delivery.config

import br.org.restaurantepopular.data_db.repository.category.CategoryDAO
import br.org.restaurantepopular.data_db.repository.category.CategoryRepository
import br.org.restaurantepopular.usecase.category.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class CategoryModule {

    @Bean
    fun getCategoryByIdUseCase(categoryRepository: ICategoryRepository) = GetCategoryByIdUseCase(categoryRepository)

    @Bean
    fun createCategoryUseCase(categoryRepository: ICategoryRepository) = CreateCategoryUseCase(categoryRepository)

    @Bean
    fun updateCategoryUseCase(categoryRepository: ICategoryRepository) = UpdateCategoryUseCase(categoryRepository)

    @Bean
    fun listCategoriesUseCase(categoryRepository: ICategoryRepository) = ListCategoriesUseCase(categoryRepository)

    @Bean
    fun deleteCategoryUseCase(categoryRepository: ICategoryRepository) = DeleteCategoryUseCase(categoryRepository)

    @Bean
    fun categoryRepository(categoryDAO: CategoryDAO) = CategoryRepository(categoryDAO)
}